src/test/java/uk/ac/ebi/embl/api/validation/submission/SubmissionValidationPlanTest.java

```diff
@@ -479,4 +482,67 @@ public void testGenomeSubmissionwitFastawithValidChromosomeListMultipleTimesInPa
 
 		CompletableFuture.allOf(futs.toArray(new CompletableFuture[futs.size()])).join();
 	}
+
+	@Test
+	public void testCovid19GenomeSize_AGP() throws ValidationEngineException, IOException {
+		String rootPath = "genome/covid19_seq_length_check/";
+		String fastaFile = "multi_entries_fasta.txt"; //29,518bp
+		String agpFile = "agp.txt";
+		options.context = Optional.of(Context.genome);
+		SubmissionFiles submissionFiles = new SubmissionFiles();
+		submissionFiles.addFile(initSubmissionFixedTestFile(rootPath,fastaFile, FileType.FASTA));
+		submissionFiles.addFile(initSubmissionFixedTestFile(rootPath,agpFile, FileType.AGP));
+		options.submissionFiles = Optional.of(submissionFiles);
+		options.reportDir = Optional.of(initSubmissionTestFile(rootPath,fastaFile, FileType.FASTA).getFile().getParent());
+		options.processDir = Optional.of(initSubmissionTestFile(rootPath,fastaFile, FileType.FASTA).getFile().getParent());
+		options.assemblyInfoEntry.get().setAssemblyType(AssemblyType.COVID_19_OUTBREAK.getValue());
+
+		clearInfoFiles(options.processDir.get());
+		SubmissionValidationPlan plan = new SubmissionValidationPlan(options);
+		plan.execute(); // should throw no exception
+	}
+
+	@Test
+	public void testCovid19GenomeSize_AGP_MaxSizeException() throws ValidationEngineException, IOException {
+		String rootPath = "genome/covid19_seq_length_check/";
+		String fastaFile = "multi_entries_fasta_over31bp.txt"; //31,711bp
+		String agpFile = "agp_over31bp.txt";
+		options.context = Optional.of(Context.genome);
+		SubmissionFiles submissionFiles = new SubmissionFiles();
+		submissionFiles.addFile(initSubmissionFixedTestFile(rootPath,fastaFile, FileType.FASTA));
+		submissionFiles.addFile(initSubmissionFixedTestFile(rootPath,agpFile, FileType.AGP));
+		options.submissionFiles = Optional.of(submissionFiles);
+		options.reportDir = Optional.of(initSubmissionTestFile(rootPath,fastaFile, FileType.FASTA).getFile().getParent());
+		options.processDir = Optional.of(initSubmissionTestFile(rootPath,fastaFile, FileType.FASTA).getFile().getParent());
+		options.assemblyInfoEntry.get().setAssemblyType(AssemblyType.COVID_19_OUTBREAK.getValue());
+
+		clearInfoFiles(options.processDir.get());
+		String expectedMsg = String.format("%s maximum genome size is %d bp.", AssemblyType.COVID_19_OUTBREAK.getValue(), GenomeUtils.COVID_19_OUTBREAK_GENOME_MAX_SIZE);
+		SubmissionValidationPlan plan = new SubmissionValidationPlan(options);
+		thrown.expect(ValidationEngineException.class);
+		thrown.expectMessage(expectedMsg);
+		plan.execute();
+	}
+
+	@Test
+	public void testCovid19GenomeSize_ChromoList() throws ValidationEngineException, IOException {
```

Comment:
- Fine to have the AGP. Perhaps might be good to test the majority failure case: testCovid19GenomeSizeOneChromsomeFailure

-  testCovid19GenomeSizeContigsScafffoldsSuccess
-  testCovid19GenomeSizeContigsScafffoldsFailure
- testCovid19GenomeSizeOneChromosomeSuccess
- testCovid19GenomeSizeOneChromosomeFailure